docs/configuration.md

```diff
@@ -0,0 +1,457 @@
+    metadata: {
+        description: "",
+        poster: ""
+    },
+    playback: {
+        audioLanguage: "",
+        textLanguage: "",
+        useNativeTextTrack: false,
+        volume: 1,
+        playsinline: true,
+        preload: "none",
+        autoplay: false,
+        allowMutedAutoPlay: true,
+        muted: false,
+        options: {
+            html5: {
+                hls: {},
+                dash: {}
+            }
+        },
+        preferNative: {
+            hls: false,
+            dash: false
+        },
+        streamPriority: [
+            {
+                engine: "html5",
+                format: "hls"
+            },
+            {
+                engine: "html5",
+                format: "dash"
+            },
+            {
+                engine: "html5",
+                format: "progressive"
+            }
+        ]
+    }
+};
+```
+## 
+>### config.logLevel
+>##### Type: `string`
+>##### Default: `"ERROR"`
+>##### Description: Defined the player log level.
+>Possible values: `"DEBUG", "INFO", "TIME", "WARN", "ERROR", "OFF"`
+## 
+>### config.type
+>##### Type: `string`
+>##### Default: `"Unknown"`
+>##### Description: Defined the media type.
+>Possible values: `"Vod", "Live", "Image", "Audio", "Unknown"`
+## 
+>### config.sources
+>##### Type: `PKSourcesConfig`
+>```js
+>{
+>  dash: Array<PKMediaSourceObject>
+>  hls: Array<PKMediaSourceObject>
+>  progressive: Array<PKMediaSourceObject>
+>}
+>```
+>>##### Type `PKMediaSourceObject`
+>>```js
+>>{
+>>  mimetype: string,
+>>  url: string,
+>>  id: string, // optional
+>>  bandwidth: number, // optional
+>>  width: number, // optional
+>>  height: number, // optional
+>>  drmData: Array<PKDrmDataObject> // optional
+>>}
+>>```
+>>##### Type `PKDrmDataObject`
+>>```js
+>>{
+>>  licenseUrl: string,
+>>  scheme: string,
+>>  certificate: string // optional
+>>}
+>>```
+>##### Default:
+>```js
+>{
+>  hls: [],
+>  dash: [],
+>  progressive: []
+>}
+>```
+>##### Description: Defined the optional sources for playback.
+>Should map media source extension type to its array of sources.
+>#### Example:
+>```js
+>var config = {
+>    sources: {
+>        hls: [
+>            {
+>                mimetype: "application/x-mpegurl",
+>                url: "https://wowzaec2demo.streamlock.net/vod-multitrack/_definst_/smil:ElephantsDream/ElephantsDream.smil/playlist.m3u8"
+>            }
+>        ],
+>        dash: [
+>            {
+>                mimetype: "application/dash+xml",
+>                url: "https://storage.googleapis.com/shaka-demo-assets/angel-one/dash.mpd"
+>            }
+>        ],
+>        progressive: [
+>            {
+>                mimetype: "video/mp4",
+>                url: "https://www.w3schools.com/html/mov_bbb.mp4"
+>            }
+>        ]
+>    }
+>};
+>```
+## 
+>### config.plugins
+>##### Type: `PKPluginsObject`
+> `{ [plugin: string]: Object }`
+>##### Default: `{}`
+>##### Description: Defined the active plugins.
```

Comment:
**Description: Defines the active plugins. This should map the plugin to its config object.**